src/objective-c/tests/InteropTests.m

```diff
@@ -457,4 +457,99 @@ - (void)testRPCAfterClosingOpenConnections {
   [self waitForExpectationsWithTimeout:TEST_TIMEOUT handler:nil];
 }
 
+- (void)testAlternateDispatchQueue {
+  XCTAssertNotNil(self.class.host);
+  __weak XCTestExpectation *expectation1 = [self expectationWithDescription:@"AlternateDispatchQueue1"];
+
+  NSArray *requests = @[@27182, @8, @1828, @45904];
+  NSArray *responses = @[@31415, @9, @2653, @58979];
+
+  // Set the default dispatch queue
+  NSString *queue1_label = @"test.queue1";
+  NSString *queue2_label = @"test.queue2";
+  dispatch_queue_t queue1 = dispatch_queue_create([queue1_label UTF8String], DISPATCH_QUEUE_SERIAL);
+  dispatch_queue_t queue2 = dispatch_queue_create([queue2_label UTF8String], DISPATCH_QUEUE_SERIAL);
+  [_service setDefaultResponseDispatchQueue:queue1];
```

Comment:
There are two ways to set the queue on which protobuf is decoded. I'd suggest to remove this way (setDefaultResponseDispatchQueue) and only give one canonical way to setting the queue. THis will reduce possible combinations and debugging complexity.